pkgs/by-name/sc/scuba/package.nix

```
@@ -0,0 +1,68 @@
+{
+  lib,
+  pkgsStatic,
+  fetchFromGitHub,
+  python3Packages,
+}:
+
+let
+  version = "2.14.0";
+
+  src = fetchFromGitHub {
+    owner = "JonathonReinhart";
+    repo = "scuba";
+    tag = "v${version}";
+    hash = "sha256-AX70js/bvt88zWJlXpuHIeBsBRfAL4qZjuthPFKSnFI=";
+  };
+
+  # This must be built statically because scuba will execute unknown docker environments
+  scubainit = pkgsStatic.rustPlatform.buildRustPackage rec {
+    pname = "scubainit";
+    inherit src version;
+
+    sourceRoot = "${src.name}/scubainit";
+
+    useFetchCargoVendor = true;
+    cargoHash = "sha256-YUYo2B5hzzmDeNiWUC+198Qbz+JPgUJfpAqyPWAXTRA=";
+  };
+in
+python3Packages.buildPythonPackage rec {
+  pname = "scuba";
+  inherit src version;
+  pyproject = true;
+
+  build-system = with python3Packages; [
+    setuptools
+  ];
+
+  dependencies = with python3Packages; [
+    argcomplete
+    pyyaml
+  ];
+
+  postPatch = ''
+    # Version detection fails
+    # Patch in the version instead
+    substituteInPlace scuba/version.py \
+    --replace-fail "__version__ = get_version()" "__version__ = \"${version}\""
+
+    # Disable calling cargo through the make file
+    # scubainit has already been built
+    substituteInPlace setup.py \
+    --replace-fail "check_call([\"make\"])" "pass"
+  '';
+
+  preBuild = ''
+    # Copy scubainit into the build tree
+    cp ${scubainit}/bin/scubainit scuba/scubainit
```

```suggestion
    ln -s ${scubainit}/bin/scubainit scuba/scubainit
```
We can safe some duplicated storage or does that not work?

---

It seems the same regardless if I copy or symlink the file. The result contains a real copy of `scubainit` so I don't think it de-duplicates anything after the build. However, it removes one copy from the build so it's worth changing anyway.